src/charts/heatmap.js

```diff
@@ -0,0 +1,350 @@
+define(function (require) {
+    'use strict';
+
+    const d3Array = require('d3-array');
+    const d3Selection = require('d3-selection');
+    const d3Scale = require('d3-scale');
+    const d3Transition = require('d3-transition');
+    const d3Interpolate = require('d3-interpolate');
+
+    const { exportChart } = require('./helpers/export');
+    const colorHelper = require('./helpers/color');
+
+    /**
+     * @typedef HeatmapData
+     * @type {Array[]}
+     * @property {Number} week
+     * @property {Number} day
+     * @property {Number} value
+     *
+     * @example
+     * [
+     *     {
+     *         day: 0,
+     *         hour: 0,
+     *         value: 7
+     *     },
+     *     {
+     *         day: 0,
+     *         hour: 1,
+     *         value: 10
+     *     }
+     * ]
+     */
+
+    /**
+     * Reusable Heatmap API module that renders a
+     * simple and configurable heatmap chart.
+     *
+     * @module Heatmap
+     * @tutorial heatmap
+     * @requires d3-array, d3-selection, d3-scale, d3-interpolate, d3-transition
+     *
+     * @example
+     * let heatmap = heatmap();
+     *
+     * heatmap
+     *     .width(500);
+     *
+     * d3Selection.select('.css-selector')
+     *     .datum(dataset)
+     *     .call(heatmap);
+     */
+    return function module() {
+
+        let margin = {
+                top: 40,
+                right: 20,
+                bottom: 20,
+                left: 40
+            },
+            width = 780,
+            height = 270,
+            svg,
+            data,
+            chartWidth,
+            chartHeight,
+
+            boxes,
+            boxSize = 30,
+            boxBorderSize = 2,
+            boxInitialOpacity = 0.2,
+            boxFinalOpacity = 1,
+            boxInitialColor = '#BBBBBB',
+            boxBorderColor = '#FFFFFF',
+
+            colorScale,
+            colorSchema = colorHelper.colorSchemas.red,
+
+            animationDuration = 2000,
+
+            dayLabels,
+            daysHuman = ['Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa', 'Su'],
+            dayLabelWidth = 30,
+
+            hourLabels,
+            hoursHuman = [
+                '00h', '01h', '02h', '03h', '04h', '05h', '06h', '07h', '08h',
+                '09h', '10h', '11h', '12h', '13h', '14h', '15h', '16h', '17h',
+                '18h', '19h', '20h', '21h', '22h', '23h'
+            ],
+            hourLabelHeight = 20,
+
+            getValue = ({value}) => value,
+
+            toremove;
```

Comment:
what does this variable represent? Is this `toRemove`? maybe a different naming could help clear this out

Reply:
Oops! Great Catch!